src/freenet/client/filter/ContentDataFilter.java

```diff
@@ -35,7 +35,17 @@
 	 * if data is merely badly formatted - any such exceptions should be
 	 * caught and converted to a DataFilterException.
 	 */
-	public void readFilter(
+	void readFilter(
 			InputStream input, OutputStream output, String charset, HashMap<String, String> otherParams,
 			String schemeHostAndPort, FilterCallback cb) throws DataFilterException, IOException;
+
+	/**
+	 * Compatibility for readFilter without schemeHostAndPort. Please use readFilter with schemeHostAndPort.
+	 */
+	default void readFilter(
+			InputStream input, OutputStream output, String charset, HashMap<String, String> otherParams,
```

Comment:
Can you use the less restrictive `Map<String, String>` as parameter type here? It shouldn’t break existing code and is clearer to read.

Reply:
that sounds good, yes. And thankfully it’s just a “migrate”-option away to move the implementing classes, too.